dwave/cloud/solver.py

```diff
@@ -739,6 +749,149 @@ def upload_cqm(self, cqm):
         return self.upload_problem(cqm)
 
 
+class NLSolver(BaseUnstructuredSolver):
+    """NL solver interface.
+
+    This class provides an :term:`NL model` sampling method and encapsulates
+    the solver description returned from the D-Wave cloud API.
+
+    Args:
+        client (:class:`~dwave.cloud.client.Client`):
+            Client that manages access to this solver.
+
+        data (`dict`):
+            Data from the server describing this solver.
+    """
+
+    _handled_problem_types = {"nl"}
```

Comment:
Almost certainly doesn't matter, but `frozenset` would be very slightly "safer"?

Reply:
Probably. Or `tuple` even would probably be better.. But we should change it on all solvers then..